server/aesdsocket.c

```diff
@@ -197,9 +425,19 @@ int main(int argc, char *argv[])
 	      break;
 	  } else if (!(connfd < 0)) {
 	      syslog(LOG_INFO, "server accept the client...\n");
-   
-	      process_client_connection(connfd);
+
+	      if (create_client_thread(connfd) == 0) {
+		  syslog(LOG_INFO, "thread successfully for %d connection", connfd);
+	      }  
 	  }
+
+	  //check if client thread is joined
+	  (void) check_for_finished_client_threads();
+       }
+
+       //control comes when signal is detected and run_connections == 0
+       while (check_for_finished_client_threads() != 0) {
```

Comment:
Nice refactor here between assignment 5 and 6 and re-use of logic.  One suggestion, you could remove the need to make `head` global if you passed it into the `check_for` functions.